Sources/MapboxMaps/Foundation/Tracing.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+import os.signpost
+import MapboxCommon
+
+private let subsystem = "com.mapbox.maps"
+
+internal extension OSLog {
+    @available(iOS 12, *)
+    private static let _poi = OSLog(subsystem: subsystem, category: .pointsOfInterest)
+    private static let _platform = OSLog(subsystem: subsystem, category: "platform")
+    private static let tracingEnabled: Bool = SettingsServiceFactory.getInstance(storageType: .nonPersistent).isTracingEnabled
+
+    static let platform: OSLog = tracingEnabled ? _platform : .disabled
+    static let poi: OSLog = {
+        if #available(iOS 12, *), tracingEnabled {
+            return ._poi
+        } else {
+            return .disabled
+        }
+    }()
+
+    func beginInterval(name: StaticString, beginMessage: String? = nil) -> SignpostInterval {
+        let interval = SignpostInterval(log: self, name: name)
+        interval.begin(message: beginMessage)
+        return interval
+    }
+
+    func signpostEvent(name: StaticString, message: String? = nil) {
+        if #available(iOS 12, *) {
+            signpost(.event, log: self, name: name, message: message)
+        }
+    }
+}
+
+internal struct SignpostInterval {
+    private enum SignpostType {
+        case begin
+        case end
+        case event
+
+        @available(iOS 12, *)
+        var asOSType: OSSignpostType {
+            switch self {
+            case .begin: return .begin
+            case .end: return .end
+            case .event: return .event
+            }
+        }
+    }
```

Comment:
Can we use `OSSignpostType` instead?

Reply:
unfortunately, no :(
Even though the type is available for iOS < 12, it's static properties like `end, begin,  event` are available only from iOS 12

Reply:
Oh, right
Pretty bad 🙁